modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -93,6 +97,11 @@ class PandasDataframe(ClassLogger):
     # These properties flag whether or not we are deferring the metadata synchronization
     _deferred_index = False
     _deferred_column = False
+    # If some logic involving materialisation is performed in loop the below property
+    # decides if the materialisation should be performed in each loop iteration.
+    # This value is set True by default. For  implementations eg Ray where this is an
+    # antipattern the value will be set to false.
+    _materialize_in_loop = True
```

Comment:
is this going to be set to `False` in a separate PR?

Reply:
It is included in this PR, got the file overwritten by mistake in the previous push, have corrected it now.https://github.com/modin-project/modin/pull/6842/files#diff-9efa37b31c87a1e27a85628a052c666e9db204aa7792113ca412627a6bb90567R44